src/pages/docs/platform/auth/capabilities.mdx

```diff
@@ -0,0 +1,897 @@
+//   'status': ['subscribe', 'history']
+// }
+```
+</Code>
+
+##### Ably Token with incompatible capabilities <a id="ably-token-error"/>
+
+If a set of capabilities are requested, and the intersection between those and the API key's capabilities is empty, then the `TokenRequest` will result in an error.
+
+Using the following example, an API key exists with the listed capabilities. If an [Ably Token](/docs/auth/token#tokens) is requested that specifies a set of capabilities, and there is no intersection between the capabilities of the issuing API key and requested token, then the token request will be rejected. In the following example, the callback will be returned with an error.
+
+<Code>
+```javascript
+  // API key capabilities:
+  {
+    'chat': ['*']
+  }
+
+  // Token request that specifies capabilities:
+  const tokenDetails = await auth.requestToken({ capability: {
+    'status': ['*']
+  }});
+```
+
+```python
+# API key capabilities:
+# {
+#   "chat": ["*"]
+# }
+
+token_details = await ably_rest.auth.request_token({
+    'capability': json.dumps({
+        {
+            "status": ["*"]
+        }
+    })
+})
+```
+
+```php
+/**
+  * API key capabilities:
+  * {
+  *   'chat': ['*']
+  * }
+  */
+
+// Token request that specifies capabilities:
+$tokenDetails = $rest
+    ->auth
+    ->requestToken(
+        ['capability' => json_encode(['status' => ['*']])]
+    );
+```
+
+```go
+// API key capabilities:
+// {
+//   "chat": ["*"]
+// }
+
+rest, _ := ably.NewREST(
+  ably.WithKey("{{API_KEY}}"))
+
+// Define the capabilities
+capabilities := map[string][]string{
+  "status": {"*"},
+}
+
+capabilitiesJSON, err := json.Marshal(capabilities)
+if err != nil {
+  log.Fatalf("Failed to marshal capabilities: %v", err)
+}
+
+// Define the token parameters
+tokenParams := &ably.TokenParams{
+  Capability: string(capabilitiesJSON),
+}
+
+// Request a token
+tokenDetails, err := rest.Auth.RequestToken(context.Background(), tokenParams)
+if err != nil {
+  log.Fatalf("Failed to request token: %v", err)
+}
+```
+
+```flutter
+// API key capabilities:
+// {
+//   'status': ['*']
+// }
+final tokenParams = ably.TokenParams(
+  capability: jsonEncode({
+    'status': ['*']
+  }),
+);
+
+final tokenDetails = await realtime.auth.requestToken(tokenParams: tokenParams);
+```
+</Code>
+
+##### Ably JWT capability determination <a id="ably-jwt"/>
+
+Capabilities are determined for [Ably JWTs](/docs/auth/token#jwt) in the following way:
+
+* The capabilities granted to an Ably JWT will be the intersection of the capabilities within the Ably JWT and the capabilities of the associated API key.
+* If the set of capabilities within the Ably JWT have no intersection with the capabilities of the API key, then an error will instead be returned.
+
+### Custom restrictions on channels <a id="custom-restrictions"/>
+
+It is possible for JWTs to contain authenticated claims for users that can be used to allow or disallow certain interactions in your channels.
```

Comment:
We should start off with a front-loaded (active) affirmation of what this feature enables the user. and why they would use it.